src/main/java/VASSAL/tools/swing/SwingUtils.java

```diff
@@ -53,115 +72,300 @@ public static AffineTransform descaleTransform(AffineTransform t) {
+      }      
     }
-
-    public boolean isControlDown(MouseEvent e) {
-      // The Command key on a Mac corresponds to Control everywhere else,
-      // but it obviously can't be called "Command" in Java; here it's
-      // called "Meta".
-      return e.isMetaDown();
+    
+    @Override
+    public boolean isSelectionToggle(MouseEvent e) {
+      // On Macs the "meta" key (Command key) is used to toggle items in and out of selections.
+      // Unless the legacy preference is set, in which case we let Control continue to have that function.
+      return macLegacy ? e.isControlDown() : e.isMetaDown();       
     }
-  }
+        
+    @Override
+    @SuppressWarnings("deprecation")
+    public KeyStroke getKeySystemToVassal (KeyStroke k) {
+      if (macLegacy) {
+        return k;
+      }
+      int modifiers = k.getModifiers();
+      if ((modifiers & InputEvent.META_DOWN_MASK) != 0) {
+        // Here we make "Meta" (Command) keystrokes from a Mac look like "Ctrl" keystrokes to Vassal
+        // We must also remove the deprecated META_MASK, or Java will end up restoring the META_DOWN_MASK.
+        modifiers &= ~(InputEvent.META_DOWN_MASK | InputEvent.META_MASK);
+        modifiers |= InputEvent.CTRL_DOWN_MASK;
+        return KeyStroke.getKeyStroke(k.getKeyCode(), modifiers, k.isOnKeyRelease());
+      }
+      else if ((modifiers & InputEvent.CTRL_DOWN_MASK) != 0) {
+        // Don't let Vassal recognize the Mac "Control" key as a "Ctrl" keystroke for keyboard commands. Because that would be bad-app-behavior, naughty naughty.
+        modifiers &= ~(InputEvent.CTRL_DOWN_MASK | InputEvent.CTRL_MASK);
+        return KeyStroke.getKeyStroke(k.getKeyCode(), modifiers, k.isOnKeyRelease());
+      }
+      else {
+        return k; 
+      }      
+    }
+    
+    @Override
+    @SuppressWarnings("deprecation")
+    public KeyStroke getKeyVassalToSystem (KeyStroke k) {
+      if (macLegacy) {
+        return k;
+      }
+      int modifiers = k.getModifiers();
+      if ((modifiers & InputEvent.CTRL_DOWN_MASK) != 0) {
+        // Here we turn any "Ctrl" keystrokes stored in Vassal into "Meta" (Command) keystrokes for use in communicating with a Mac
+        // We must also remove the deprecated CTRL_MASK, or Java will end up restoring the CTRL_DOWN_MASK
+        modifiers &= ~(InputEvent.CTRL_DOWN_MASK | InputEvent.CTRL_MASK);
+        modifiers |= InputEvent.META_DOWN_MASK;
+        return KeyStroke.getKeyStroke(k.getKeyCode(), modifiers, k.isOnKeyRelease());
+      } 
+      else {
+        return k; 
+      }      
+    }          
+  } 
 
   private static final InputClassifier INPUT_CLASSIFIER =
     SystemUtils.IS_OS_MAC_OSX ?
       new MacInputClassifier() : new DefaultInputClassifier();
 
-  /*
-   * @return whether the event is effectively for the left button
+  /**
+   * @return whether the event is effectively for the left button. Deprecated in favor of isVanillaLeftButtonDown()
```

Comment:
Can write it like this, then it's going to be a proper reference to the method and all the tools will easily find it and allow navigating from this javadoc right into the method, and also rename this reference if the method is renamed:

  /**
   * @return whether the event is effectively for the left button.
   * 
   * @deprecated Deprecated in favor of {@link #isVanillaLeftButtonDown(MouseEvent)}
   */